dev-support/docker/Dockerfile

```diff
@@ -0,0 +1,110 @@
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Dockerfile for installing the necessary dependencies for building Hadoop.
+# See BUILDING.txt.
+
+FROM ubuntu:20.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+WORKDIR /root
+
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+#####
+# Disable suggests/recommends
+#####
+RUN echo APT::Install-Recommends "0"\; > /etc/apt/apt.conf.d/10disableextras
+RUN echo APT::Install-Suggests "0"\; >>  /etc/apt/apt.conf.d/10disableextras
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV DEBCONF_TERSE true
+
+
+# hadolint ignore=DL3008
+###
+# Update and install common packages
+###
+RUN apt -q update \
+   && apt install -y software-properties-common apt-utils apt-transport-https ca-certificates \
+   && add-apt-repository -y ppa:deadsnakes/ppa \
+   && apt-get -q install -y --no-install-recommends \
+        bash-completion \
+        build-essential \
+        bzip2 \
+        curl \
+        docker.io \
+        g++ \
+        gcc \
+        git \
+        gnupg-agent \
+        rsync \
+        sudo \
+        vim \
+        locales \
+        wget \
+        time \
+        openjdk-8-jdk \
+        python3-setuptools \
+        python3-pip \
+        python3.5 \
+        python3.6 \
+        python3.7 \
+        python2.7 \
+        virtualenv \
+        tox
+
+###
+# Set the locale ( see https://stackoverflow.com/a/28406007/114196 )
+###
+RUN sed -i '/en_US.UTF-8/s/^# //g' /etc/locale.gen && \
+    locale-gen
+ENV LANG en_US.UTF-8
+ENV LANGUAGE en_US:en
+ENV LC_ALL en_US.UTF-8
+
+###
+# Set Python3.6 as default
+###
+RUN alias python=python3.6
+
+###
+# Install grpcio-tools mypy-protobuf for `python3 sdks/python/setup.py sdist` to work
+###
+RUN pip3 install grpcio-tools mypy-protobuf
```

Comment:
What happens without these? If they are necessary for a build environment we should add them to https://beam.apache.org/contribute/#prerequisites as well

Reply:
I'm not sure. I copied this from https://github.com/apache/beam/pull/12837

Reply:
I can confirm that without this the build fails around messages involving `sdist`.
I'm no python expert but it seems this is really needed.
So I'm leaving this in.

I've added a bit to the mentioned documentation page.